src/bmp.c

```diff
@@ -40,7 +40,7 @@ static void initialize_bmp_header(BmpObject *image) {
   image->header.size = image->header.data + image->win_header.length;
 }
 
-static void initialize_bmp_body(BmpObject *image, uint32_t image_padding, uint32_t mcu_width, short *MCU_buffer) {
+static void initialize_bmp_body(BmpObject *image, uint32_t image_padding, uint32_t mcu_width, short *MCU_buffer) { //CHANGING BUFF TO SHORT CAUSES ERRORS
```

Comment:
These "errors" are what you are supposed to fix
The data coming from the DPU is an array of shorts. Casting everything to char doesn't change how the data is stored, which is what we are trying to do.